timApp/tide_support/routes.py

```diff
@@ -0,0 +1,126 @@
+from authlib.integrations.flask_oauth2 import current_token
+from flask import Response, request
+
+from timApp.auth.oauth2.models import Scope
+from timApp.auth.oauth2.oauth2 import require_oauth
+
+from timApp.tide_support.utils import (
+    user_ide_courses,
+    ide_tasks,
+    ide_task_by_id,
+    ide_task_folders_by_doc,
+    ide_submit_task,
+)
+from timApp.user.user import User
+from timApp.util.flask.responsehelper import json_response
+from timApp.util.flask.typedblueprint import TypedBlueprint
+
+ide = TypedBlueprint("ide", __name__, url_prefix="/ide")
+
+
+@ide.get("ideCourses")
+@require_oauth(Scope.user_tasks.value)
+def get_user_ide_courses() -> Response:
+    """
+    Get all courses that the user has bookmarked and have ideCourse tag
+    :return: JSON response with all courses and their task folders
+    """
+
+    user: User = current_token.user
+    return json_response(user_ide_courses(user=user))
```

Comment:
Tyyli: `user=user` on tässä lyhyemmin `user`

```suggestion
    return json_response(user_ide_courses(user))
```

Reply:
Tämä pätee muihinkin kohtiin, jossa muotoa `key=value` ei ole pakko kayttää

Reply:
Osa korjattu. Jätin paikalleen muutamaan kohtaan, missä parametreja oli enemmän, näytti minusta näin selkeämmältä. Korjataan, jos tämä on väärin.